pb_decode.c

```diff
@@ -1184,8 +1184,8 @@ static void pb_release_single_field(const pb_field_iter_t *iter)
                 count = iter->pos->array_size;
             }
         }
-        
-        if (pItem)
+
+        if (pItem && PB_ATYPE(type) != PB_ATYPE_CALLBACK)
```

Comment:
This check could perhaps be moved up to line 1161, because the lines in between have no effect for callbacks.

Reply:
I first tried to put this check as precondition for the whole function `if (PB_ATYPE(type) != PB_ATYPE_CALLBACK) return;`, as how I reasoned, releasing a callback field doesn't ever seem to make sense. But it seems that caused some tests to fail, so was hoping you could give better comment where to do this check properly.

Reply:
Yeah. It's annoyingly delicate to get all the cases right here. Extension fields have "CALLBACK" type, but they contain a pointer to a field, which may be a submessage, which may contain pointer fields. That's why it would fail tests if you return at the beginning of the function.

Reply:
Do you think handling this on case 1161, would be the best way then?

Reply:
Yeah, I think it is. I can of course change that myself also, I'm just not currently at a machine where I can run the test set.